services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/ScanEnhancedRequest.java

```diff
@@ -203,6 +231,44 @@ public Builder limit(Integer limit) {
             return this;
         }
 
+        /**
+         * For a parallel Scan request, Segment identifies an individual segment to be scanned by an application worker.
+         * <p>
+         * <b>Note:</b>Segment IDs are zero-based, so the first segment is always 0. For example, if you want to use four
+         * application threads to scan a table or an index, then the first thread specifies a Segment value of 0, the second
+         * thread specifies 1, and so on.
+         *
+         * The value of {@link Page#lastEvaluatedKey()} returned from a parallel Scan request must be used as
+         * {@link #exclusiveStartKey} with the same segment ID in a subsequent Scan operation.
+         *
+         * The value for Segment must be greater than or equal to 0, and less than the value provided for TotalSegments.
+         *
+         * If you provide Segment, you must also provide <em>TotalSegments</em>.
+         *
+         * @param segment  identifies an individual segment to be scanned
+         * @return a builder of this type
+         */
+        public Builder segment(Integer segment) {
+            this.segment = segment;
+            return this;
+        }
+
+        /**
+         * For a parallel Scan request, TotalSegments represents the total number of segments into
+         * which the Scan operation will be divided.
+         * <p>
+         * <b>Note:</b>If you specify a TotalSegments value of 1, the Scan operation will be sequential rather than parallel.
```

Comment:
From the SDK user's perspective, isn't the scan always sequential and they need to do the parallelization?

Reply:
The user need to make parallel calls from multiple threads for each segment.  However, within each segment, the calls has to be sequential as they are paginated. 

This doc is also copied from underlying client [ScanRequest](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/dynamodb/model/ScanRequest.html#totalSegments--) model.

Reply:
Maybe we can just write: `<b>Note:</b> If you do not specify this value, the TotalSegments is effectively 1.` It seems worth making it clear that this does not need to be specified, and if you do specify it, don't bother making it 1.

I don't like that the current language implies that setting a TotalSegments > 1 makes the scan operation parallel. It's certainly A WAY to make the scan operation parallel, but the customer still has to do that work themselves. With the current wording, someone might be justified setting TotalSegments = 2 and then expect the SDK to handle the parallelism for them.

Reply:
Agreed. I made changes to reflect that fact that caller has to create separate each thread to achieve parallelism.